runelite-mixins/src/main/java/net/runelite/mixins/RSClientMixin.java

```diff
@@ -676,4 +671,43 @@ public static void menuOpened(int opened)
 		event.setMenuEntries(client.getMenuEntries());
 		eventBus.post(event);
 	}
+
+	@Inject
+	@Override
+	public boolean hasHintArrow()
+	{
+		return client.getHintArrowTargetType() != 0;
+	}
+
+	@Inject
+	@Override
+	public void clearHintArrow()
+	{
+		client.setHintArrowTargetType(0);
+	}
+
+	@Inject
+	@Override
+	public void setHintArrow(Actor actor)
+	{
+		if (actor instanceof Player)
+		{
+			client.setHintArrowTargetType(10);
+			client.setHintArrowPlayerTargetIdx(actor.getIndex());
+		}
+		else
+		{
+			client.setHintArrowTargetType(1);
+			client.setHintArrowNpcTargetIdx(actor.getIndex());
+		}
```

Comment:
```java
client.setHintArrowTargetType(actor instanceof Player ? 10 : 1);
client.setHintArrowPlayerTargetIdx(actor.getIndex());
```

Reply:
I would make a HintArrowType enum in the api for this.